File: Excel/587 Challenge.R
library(tidyverse)
library(readxl)
library(purrr)

path = "Excel/587 Repeat Vowels Order of Occurrence.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10")

process_string <- function(string) {
  tibble(char = str_split(string, "")[[1]]) %>%
    mutate(is_vowel = char %in% c("a", "e", "i", "o", "u"),
           count = cumsum(is_vowel) * is_vowel + (!is_vowel)) %>%
    mutate(char2 = str_dup(char, count)) %>%
    summarise(result = str_c(char2, collapse = "")) %>%
    pull(result)
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Strings, process_string))

all.equal(result$`Answer Expected`, test$`Answer Expected`)
#> [1] TRUE

File: Excel/500-599/587/587 Challenge.py
import pandas as pd

path = "587 Repeat Vowels Order of Occurrence.xlsx"
input = pd.read_excel(path, usecols="A", nrows=10)
test = pd.read_excel(path, usecols="B", nrows=10)

def repeat_vowels(word):
    vowels = "aeiou"
    vowel_count = 0
    return ''.join(char * (vowel_count := vowel_count + 1) if char in vowels else char for char in word)

result = input["Strings"].apply(repeat_vowels)

print(result.equals(test["Answer Expected"])) # True